phoenixconnect/phoenixconnect-registers.json

```diff
@@ -0,0 +1,188 @@
+{
+    "className": "Phoenix",
+    "protocol": "TCP",
+    "endianness": "LittleEndian",
+    "enums": [
+        {
+            "name": "ErrorCode",
+            "values": [
+                {
+                    "key": "NoError",
+                    "value": 0
+                },
+                {
+                    "key": "CableRejected13or20A",
+                    "value": 1
+                },
+                {
+                    "key": "CableRejected13A",
+                    "value": 2
+                },
+                {
+                    "key": "InvalidPPValue",
+                    "value": 3
+                },
+                {
+                    "key": "InvalidCPValue",
+                    "value": 4
+                },
+                {
+                    "key": "ChargingUnavailable",
+                    "value": 5
+                },
+                {
+                    "key": "Lock",
+                    "value": 6
+                },
+                {
+                    "key": "Unlock",
+                    "value": 7
+                },
+                {
+                    "key": "ErrorDuringLocking",
+                    "value": 8
+                },
+                {
+                    "key": "OverloadProtection",
+                    "value": 9
+                },
+                {
+                    "key": "MeterCommunicationError",
+                    "value": 10
+                },
+                {
+                    "key": "VehicleRejected",
+                    "value": 11
+                },
+                {
+                    "key": "ContactorFault",
+                    "value": 12
+                },
+                {
+                    "key": "VehicleCPDiodeMissing",
+                    "value": 13
+                },
+                {
+                    "key": "Unkown",
+                    "value": 14
+                },
+                {
+                    "key": "DCErrorCurrent",
+                    "value": 15
+                }
+            ]
+        }
+    ],
+    "blocks": [
+        {
```

Comment:
Block with a single register does not make sense, I assume you want to move `chargingTime` before firmwareVersion, and also the error code could be read in one call. I guess you split them up because firmware you need only once, and the rest has update schedule. You are not forced to stick on those schedule tpes, they are only for convinience, you can also type your own update method. Better less reading calls than preventing to re-read a property which has for sure not changed like firmware version.

Reply:
ps: if you run into the bug that blocks must be defined (just define an empty array)